### DISCUSSION TITLE
Raw dump of the database

### DISCUSSION THREAD

**Discussion by ttsirkia** [electric-sql/pglite#60] [Ideas]
If I understood correctly, there is no way to interact with the raw data of the in-memory database. Would it be possible to import and export the DB in some way in browser environment as a byte array which could be downloaded, for example?

**Comment by samwillis** [comment]
It's absolutely on the list to have some sort of method to dump the database. Not sure exactly what this will look like yet. Making the standard Postgres dump tools available I think should be doable.

**Comment by ttsirkia** [comment]
Compated to SQLite, exporting/importing is not that easy as PostgreSQL has many files and directories to store all data instead of a single file. Of course, dumping the data as SQL statements (with the dumping tools) could be one option and would prevent the issues of raw binary data which might not be compatible between versions.
